db/init.sql

```diff
@@ -4,13 +4,154 @@ CREATE DATABASE mm_db;
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
 CREATE TYPE user_role AS ENUM ('admin', 'teacher', 'student');
+CREATE TYPE attempt_state AS ENUM ('submitted', 'graded');
+-- NOTE(nrydanov): Need to think of other types together
+CREATE TYPE block_type AS ENUM ('task', 'markdown');
+
+CREATE TABLE unit_type (
+    id uuid PRIMARY KEY,
+    name varchar(64)
+);
 
 CREATE TABLE users (
-    id uuid NOT NULL PRIMARY KEY,
+    id uuid PRIMARY KEY,
+    -- Auth data
+    username varchar(32) UNIQUE NOT NULL,
     email varchar(128) UNIQUE NOT NULL,
-    name varchar(32) NOT NULL,
-    surname varchar(32) NOT NULL,
-    patronymic varchar(32),
-    role user_role NOT NULL,
-    password bytea NOT NULL
+    password bytea NOT NULL,
+    -- Identity data
+    name varchar(64) NOT NULL,
+    surname varchar(64) NOT NULL,
+    date_of_birth date,
+    -- Service info
+    created_at timestamp NOT NULL,
+    last_online timestamp
+);
+
+CREATE TABLE unit (
+    id uuid PRIMARY KEY,
+    name varchar(128) NOT NULL,
+    unit_type uuid REFERENCES unit_type(id),
+    parent_id uuid REFERENCES unit(id),
+    owner_id uuid REFERENCES users(id)
+);
+
+CREATE TABLE block (
+    id uuid PRIMARY KEY,
+    block_type block_type NOT NULL,
+    data jsonb NOT NULL,
+    course_id uuid NOT NULL -- REFERENCES course(id)
```

Comment:
Why is reference commented out?

Reply:
Don't remember, actually. I'll look up for it a bit later.